File: Practice/src/coding_basic_training/MakeItSquare.js
function solution(arr) {
    var answer = [];
    var max = Math.max(arr.length, arr[0].length);
    for (let i = 0; i < max; i++) {
        let temp = Array(max).fill(0);
        for (let j = 0; i < arr.length && j < arr[i].length; j++) {
            temp[j] = arr[i][j];
        }
        answer.push(temp);
    }
    return answer;
}

File: Practice/src/coding_basic_training/MakeItSquare.py
def solution(arr):
    n = max(len(arr), len(arr[0]))
    if len(arr) <= len(arr[0]):
        for _ in range(n - len(arr)):
            arr.append([0] * n)
        return arr
    for i in arr:
        for _ in range(n - len(i)):
            i.append(0)
    return arr
